src/msspreadsheets/spreadsheet_endpoint.bal

```
@@ -0,0 +1,639 @@
+            refreshConfig: {
+                clientId: msGraphConfig.msClientID,
+                clientSecret: msGraphConfig.msClientSecret,
+                refreshToken: msGraphConfig.msRefreshToken,
+                refreshUrl: msGraphConfig.msRefreshURL,
+                clientConfig: {
+                    secureSocket: {
+                        trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                        }
+                    }
+                }
+            }
+        });
+
+        http:BearerAuthHandler oauth2Handler3 = new (oauth2Provider3);
+
+        self.workbookClient = new (msGraphConfig.baseUrl, {
+            auth: {
+                authHandler: oauth2Handler3
+            },
+            secureSocket: {
+                trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                }
+            }
+        });
+    }
+
+    # Get the properties of the workbook.
+    # + return - Properties of the Workbook
+    public function getProperties() returns WorkbookProperties {
+        return self.properties;
+    }
+
+    # Open a worksheet on this workbook.
+    # + worksheetName - name of the worksheet to be opened
+    # + return - A Worksheet client object on success, else returns an error
+    public remote function openWorksheet(string worksheetName) returns @tainted (Worksheet|error) {
+        http:Request request = new;
+        http:Response|error response = self.workbookClient->get("/v1.0/me/drive/root:" + self.properties.path +
+                            self.properties.workbookName + ".xlsx:/workbook/worksheets/" + worksheetName, request);
+
+        if response is error {
+            HttpError httpError = error(HTTP_ERROR, message = "Error occurred while accessing the Microsoft Graph API.", 
```

Shall we just return the error here? Check the other places as well.

---

Ok, Done